call_apply_bind_exercise/callApplyBind.js

```diff
@@ -0,0 +1,68 @@
+var obj = {
+    fullName: "Harry Potter",
+    person: {
+        sayHi: function(){
+            return "This person's name is " + this.fullName
+        }.bind(obj)
+    }
+}
+
+
+function sumEvenArguments() {
+  var args = [].slice.call(arguments);
+
+  return args.reduce(function(acc, nextValue) {
+    if( nextValue % 2 === 0 ) acc += nextValue;
+    return acc;
+  }, 0);
+
+  // return args.filter(function(a) { return (a % 2) === 0; }).reduce(function(acc, nextValue) {
+  // 	return acc + nextValue
+  // }, 0);
+}
+
+function arrayFrom() {
+  return [].slice.call(arguments);
+}
+
+function invokeMax(fn, num) {
+  var count = num;
```

Comment:
If you count down, you don't even need to copy the variable.  Numbers are not reference types.